app/screens/channel/channel_nav_bar/channel_title/channel_title.js

```diff
@@ -43,9 +49,54 @@ export default class ChannelTitle extends PureComponent {
     }
 
     render() {
-        const {currentChannelName, displayName, isChannelMuted, onPress, theme} = this.props;
-        const channelName = displayName || currentChannelName;
+        const {currentChannelName, displayName, channelType, isChannelMuted, onPress, theme, isGuest, hasGuests, canHaveSubtitle} = this.props;
+
         const style = getStyle(theme);
+
+        const channelName = displayName || currentChannelName;
+        let hasGuestsText = null;
```

Comment:
```js
const hasGuestText = this.renderHasGuest(styles);
```

then the function would be
```js
renderHasGuest = (styles) => {
    const {canHaveSubtitle, channelType, isGuest) = this.props;
    if (!isGuest || !canHaveSubtitle) {
        return null;
    }

    let hasGuestText;
    switch (channelType) {
    case General.DM_CHANNEL:
        hasGuestText = (
            <FormattedText
                id='channel.isGuest'
                numberOfLines={1}
                ellipsizeMode='tail'
                defaultMessage='This person is a guest'
                style={style.guestsText}
            />
        );
        break;
    case General.GM_CHANNEL:
        hasGuestText = (
            <FormattedText
                id='channel.hasGuests'
                numberOfLines={1}
                ellipsizeMode='tail'
                defaultMessage='This group message has guests'
                style={style.guestsText}
            />
        );
        break;
    default:
        hasGuestText = (
            <FormattedText
                id='channel.channelHasGuests'
                numberOfLines={1}
                ellipsizeMode='tail'
                defaultMessage='This channel has guests'
                style={style.guestsText}
            />
        );
    }

    return (
        <View style={style.guestsWrapper}>
            {hasGuestText}
        </View>
    );
};
```

Reply:
I changed it in other way, I simplified the code but I didn't generate a reusable component because it used in two different places, and are not using exactly the same thing, so... I think for now is cleaner a bit of duplication.